include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -336,30 +336,31 @@ __parallel_transform_reduce(_ExecutionPolicy&& __exec, _Up __u, _Cp __combine, _
     __wgroup_size = ::std::min(__wgroup_size, oneapi::dpl::__internal::__kernel_work_group_size(
                                                   ::std::forward<_ExecutionPolicy>(__exec), __kernel));
 #endif
-    auto __mcu = oneapi::dpl::__internal::__max_compute_units(::std::forward<_ExecutionPolicy>(__exec));
-    auto __n_groups = (__n - 1) / __wgroup_size + 1;
-    __n_groups = ::std::min(decltype(__n_groups)(__mcu), __n_groups);
-    // TODO: try to change __n_groups with another formula for more perfect load balancing
+    // It can be replaced with another value for more perfect load balancing
+    auto __iters_per_witem = decltype(__wgroup_size)(4);
+    auto __size_per_wg = __iters_per_witem * __wgroup_size; // how much buffer elements work group processes
+    auto __n_groups = (__n - 1) / __size_per_wg + 1;        // number of work groups
+    auto __n_blocks = (__n - 1) / __iters_per_witem + 1;    // number of bunch of elements that the algorithm processes
```

Comment:
maybe `__n_blocks` -> `__n_items`?

Regarding a formula. It would be great, but it's mathematically incorrect, e.g.
```
__n = 10;
__iters_per_witem = 1;
__wgroup_size = 2;
```

Reply:
You right, the proposed formula is wrong. It works only if `__n % __wgroup_size == 0`. 

With fact above `__n_blocks ` is not a global number of work items, but a number of items remained after reduction by initialization stage (it performs reduction before filling local memory during the first run).

`__n_items` is better than `__n_blocks` or `__global_size`. Let's use it.